File: 021_Merge_Two_Sorted_Lists/021_1.java
class Solution 
{
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) 
    {
        ListNode ans = new ListNode(0);
        ListNode dummy = ans;
        while (l1 != null && l2 != null)
        {
            if (l1.val <= l2.val)
            {
                ans.next = l1;
                l1 = l1.next;
            }
            else
            {
                ans.next = l2;
                l2 = l2.next;
            }
            ans = ans.next;
        }
        ans.next = l1 != null? l1 : l2;
       /* if (l1 != null)
            ans.next = l1;
        else
            ans.next = l2; */
        return dummy.next;
    }
}

File: 021_Merge_Two_Sorted_Lists/021_1.py
class Solution:
    def mergeTwoLists(self, l1, l2):
        if None in (l1, l2):
            return l1 or l2

        dummy = cur = ListNode(0)
        while l1 and l2:
            if l1.val <= l2.val:
                cur.next = l1
                l1 = l1.next
            else:
                cur.next = l2
                l2 = l2.next
            cur = cur.next
        cur.next = l1 or l2
        return dummy.next